test/scenarios/bookshop/read.test.js

```diff
@@ -360,6 +360,96 @@ describe('Bookshop - Read', () => {
     expect(res.data.value.length).to.be.eq(2)
   })
 
+  test('Filter Books(complex filter in apply) and groupby(by navigation sub properties) that return no records', async () => {
+    // This test is cover next issue https://github.com/cap-js/cds-dbs/issues/1228
+    // We have apply filter that did not match any records.
+    // Also we have group by navigation sub property.
+    // And $count is true.
+    // Expeted result is empty array
+    const res = await GET(`/admin/Books?$apply=filter(price le -999)/groupby((author/name,author/ID))&$top=500&$count=true`,
+      admin//
+    );
+    expect(res.status).to.be.eq(200)
+    expect(res.data["@odata.count"]).to.be.eq(0);
+    expect(res.data.value.length).to.be.eq(0)
+  })
+
+  test('groupby(by navigation sub properties) that return at least some records', async () => {
+    // This test is cover next issue https://github.com/cap-js/cds-dbs/issues/1228
+    // We have apply filter - it should return any records
+    // Also we have group by navigation sub property.
+    // And $count is true.
+    const res = await GET(`/admin/Books?$apply=filter(price ge -1)/groupby((author/name,author/ID))&$top=500&$count=true`,
+      admin
+    );
+    expect(res.status).to.be.eq(200);
+    expect(res.data.value.length).to.be.gt(0);
+    expect(res.data["@odata.count"]).to.be.gt(0);
+
+    expect(
+      res.data.value.every(
+        item =>
+          // Lets validate that we have {author:{ID:number, name:string}}
+          'author' in item &&
+          'ID' in item.author &&
+          'name' in item.author
+      ),
+    ).to.be.true
+  })
+
+  test('Run SELECT without Columns but with group by', async () => {
+    const queryAllBooks = {
+      SELECT: {
+        from: { ref: ['sap.capire.bookshop.Books'], as: 'Books' },
+      }
+    }
+    const resultAllBooks = await cds.db.run(queryAllBooks);
+    expect(resultAllBooks.length).to.be.gt(0); // Make sure that we have any books.
+
+    const queryGropuByAuthors = {
+      SELECT: {
+        from: { ref: ['sap.capire.bookshop.Books'], as: 'Books' },
+        groupBy: [{ ref: ['author', 'name'] }, { ref: ['Books', 'author_ID'] }]
+      }
+    }
+
+    let resultGroupByAuthors = null;
+    let handleError = false;
+    // HANA and Postgres did not allow to have select with coulmns that are not agregate or group by.
+    // On other hand - sqlite allow it. TODO: disable this posibility at sqlite level ?
```

Comment:
I'd say this is expected (SQlite takes some things more easy than other dbs):

Every column that appears in the SELECT list must either
• be in the GROUP BY, or
• be wrapped in an aggregate function.

→ With CAP’s query builder, always give an explicit projection when you use GROUP BY; otherwise the implicit * drags in extra columns and the statement becomes invalid.

we may do some auto group-by stuff in the feature. But for now, this test can be removed IMO.